src/main/java/org/tugraz/sysds/common/Types.java

```diff
@@ -38,14 +38,17 @@
 	 * Data types (tensor, matrix, scalar, frame, object, unknown).
 	 */
 	public enum DataType {
-		TENSOR, MATRIX, SCALAR, FRAME, LIST, UNKNOWN;
+		FEDERATED, TENSOR, MATRIX, SCALAR, FRAME, LIST, UNKNOWN;
```

Comment:
What does this new type, FEDERATED mean? Only tensors can be federated or anything from the matrix to list to tensor?

Reply:
Since we treat `FederatedObject` differently (we want to call different methods than for example for matrices) I decided to add a new Type. Currently `FEDERATED` refers to an `FederatedObject` which is a `CacheableData<MatrixBlock>`, and it will be generalized for `TensorBlocks` at some point. Does this explain the meaning?

Reply:
Ok I talked to Matthias and I will remove this new type and do it another way, I will write a comment once I have done that.